storage/zynq7000-flash/flashdrv.c

```diff
@@ -414,47 +411,46 @@ static ssize_t flashdrv_blkWrite(struct _storage_t *strg, off_t start, const voi
 {
 	ssize_t res;
 	size_t sectSz;
+	off_t regOffs;
 	unsigned int sectID, regID;
 
 	size_t chunkSz = 0, freeSz = 0, saveSz = 0;
 
-	if (strg == NULL || strg->dev == NULL || (start + size) > strg->size || data == NULL) {
```

Comment:
I assume we depend on this params being checked outside?

Reply:
Yes, they should be checked by the upper layer